### DISCUSSION TITLE
How much contents is too much under a single content definition?

### DISCUSSION THREAD

**Discussion by vengi83644** [OrchardCMS/OrchardCore#10753] [Q&A]
We are using the Decoupled CMS site.

This discussion is for the Glossary on our website. We have added a new top-level content type Glossary and next-level content type Glossary items.

The glossary type is the main type that would hold all the glossary items and then list them on our website. We are having around **1500 glossary items** which makes the content definition page in the CMS admin too slow. It's taking around 3-5 mins to load the page.

Also for publishing, it takes around 5-10 mins.

But for getting the contents using the method, GetContentItemByHandleAsync is quick approx 5 seconds.

Do we have a benchmark or any applicable limit for the content types? Or can anyone suggest a solution for Glossary?

I see Glossary in OrchardCore documentation. That holds minimal content.

Thanks,
R. Venkatesan

**Comment by jtkech** [comment]
Looks like you created too much content types definitions, we can have hundreds of thousands content items but based on much less shared content type definitions.

Note: With our default setup recipes we enable the `FileContentDefinition` feature, so content type definitions are persisted in a file by default, but if you use a recipe that doesn't enable this feature, definitions are persisted in the database.
